src/web_peer_connection.cpp

```diff
@@ -411,7 +411,16 @@ void web_peer_connection::write_request(peer_request const& r)
 			auto redirection = m_web->redirects.find(f.file_index);
 			if (redirection != m_web->redirects.end())
 			{
-				request += redirection->second;
+				auto const& redirect = redirection->second;
+				if (using_proxy && !redirect.empty() && (redirect[0] == '/'))
```

Comment:
"m_url" will contain "http://example.com/" and in http proxy case will be added to "request" variable. if "redirection->second" contains "/redirect_file" then "request" may contain "http://example.com//redirect_file".